File: 프로그래머스/lv1/72410. 신규 아이디 추천/신규 아이디 추천.js
function solution(new_id) {
    var answer = (new_id.toLowerCase()
        .replace(/[^0-9a-z\-\.\_]/g, '')
        .replace(/\.+/g,'.')
        .replace(/^\.|\.$/, '')||'a')
        .substr(0, 15)
        .replace(/\.$/, '');
    const len = answer.length;
    if(len < 3){
        answer += answer[len-1].repeat(3-len);
    }
    return answer;
}

File: 프로그래머스/lv1/72410. 신규 아이디 추천/신규 아이디 추천.java
class Solution {
    public String solution(String new_id) {
        String answer = new_id;
        //step 1
        answer = answer.toLowerCase()
            .replaceAll("[^a-z|0-9|\\-|\\_|\\.]", "")
            .replaceAll("\\.+", ".")
            .replaceAll("^[.]|[.]$","");
        //step 5
        if(answer.equals("")) answer = "a";
        //step 6
        if(answer.length()>=16) answer = answer.substring(0,15).replaceAll("[.]$","");
        //step 7
        if(answer.length()==1)
            answer += answer + answer;
        else if(answer.length()==2)
            answer += answer.substring(1,2);
        return answer;
    }
}